### PR TITLE
#2933 - Fix: Forum page with group associated title tag issue

### LINKED ISSUE #2933: Forum page with group associated title tag issue
**Describe the bug**
When the forum page is associated with the group, there's an issue with the title tag.

**To Reproduce**
1. Create a forum and associate it with the group.
2. Check the forum page.
3. You'll see that title gets weird.

**Expected behavior**
The title should be normal as what it is when the forum has no group associated.

**Screenshots**
Forum with no group: https://prnt.sc/1rtkptv
Forum with group: https://prnt.sc/1rtkptv

**Environment**
Which site did you replicate this issue on? Demo site

**Support ticket links**
https://secure.helpscout.net/conversation/1621712112/162415?folderId=4222125

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2933 .

### How to test the changes in this Pull Request:

1. Create a forum and associate it with the group.
2. Check the forum page.
3. You'll see that title gets like Discussion - Group - Site Name.

Related Issue solved:
1. Copy the forum with no group and discussion URL
2. Paste in the activity feed
3. Can see the live preview is working fine instead of https://prnt.sc/1rtktn4

### Proof Screenshots or Video
https://vimeo.com/645160371/e987c33080

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]

**Comment by sumitbuddyboss** [comment]
Hello @milanhirapra 

Issue >

Getting the same issue, still the title is not updated. Check the below screenshot.
https://prntscr.com/21dyaqb

Thank you

**Comment by milanhirapra** [comment]
Hello @sumitbuddyboss,

I have fixed this issue. Here is the video: https://vimeo.com/653974721/2469e53382

CC: @KartikSuthar @chetansatasiya

**Review by chetansatasiya — DISMISSED** [review]

File: src/bp-forums/classes/class-bp-forums-component.php [hunk 1]
```diff
@@ -123,6 +123,8 @@ public function setup_actions() {
 
 			// Setup the components
 			add_action( 'bp_init', array( $this, 'setup_components' ), 7 );
+			// Setup meta title.
+			add_filter( 'pre_get_document_title', array( $this, 'bb_group_forums_set_title_tag' ), 999, 1);
```
Comment (KartikSuthar) [inline src/bp-forums/classes/class-bp-forums-component.php:127]:
```suggestion
			add_filter( 'pre_get_document_title', array( $this, 'bb_group_forums_set_title_tag' ), 999, 1 );
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Comment by sumitbuddyboss** [comment]
Hello @milanhirapra 

A related both the issue is resolved, 

1. Forum page with group associated title tag issue is resolve for the proof check below video.

**Cases:**

- Verify that title tag is changed or not ? _ Pass
- Vefify after changes slug of forum - Pass
- Verify after change of permalink - Pass

**Video proof:**
https://www.loom.com/share/ec5f00bafb0f4953b6afc732d80cbf16


2. The live preview is working fine with the forum url, proof check below video.

**Video proof:**
https://www.loom.com/share/b2e78eb311cf49d7a279c9c773674c5c


Test status: **Pass**

Ready for release 🚀 

Thank you

**Comment by basil-buddyboss** [comment]
Hello @milanhirapra 

This has been tested in the QA-approved branch and there were no issues encountered.

**Result: Pass**

**Comment by nickchomey** [comment]
I'm glad to see that this was fixed, but may I ask why my feedback was ignored? Specifically, to add "- Discussions - " to the title tag?

A group can have various other tabs, be they courses, events, a knowledge base, etc... It is quite possible that these might have the same title as a forum discussion topic (e.g. A discussion related to an event or knowledgebase article). So it is necessary to include Discussions (or whatever the user has set topics to be called) in the title tag in order to distinguish them. And even if there aren't any matching topics/events/kb articles etc..., it is still useful to show the group "component" in the tag - the tag isn't just for when viewing the topic, but also for SEO purposes. 

It would be very simple to add using the code I shared in my above comment.

Another very useful addition would simply be adding a filter on the return, so that we can modify/override the default tag structure ourselves. Some might want to reverse the order or add/remove fields.

More generally, filters should be added as much as possible throughout the BB code to allow for customization. 

@mikejyoung

**Comment by mikejyoung** [comment]
@nickchomey 

This PR was to fix a specific bug.

Bugs can be worked on independently by our developers, as long as they are resolving predefined logic. When we need to start changing the logic or adding new filters, UI or settings, the request becomes a suggestion and needs contribution from more team members. 

We're currently in the design and planning stage of our forum update and I'll make sure to account for your suggestions while we're designing and scoping the work. However, we do have some forum improvements coming alongside BuddyBoss Theme 2.0. 

I totally agree about filters and customization. One of our major focuses of 2021/2022 has, and is, going through each of the BuddyBoss components (forums, groups, profiles, etc) and making the code more extendable and friendly to developers (including our own!). This is lengthly work as there is still a lot to be done with the code we inherited from BuddyPress and BBPress to unlock all the great suggestions that you and all our customers are bringing up.

**Comment by nickchomey** [comment]
@mikejyoung thanks for the thoughtful response! I look forward to the progress you make with it all. 

However, it seems to me that my request/suggestion for this fix was a) part of fixing the bug, since the goal was to make the title tag functional while viewing the topic and with SEO and b) completely insignificant and could have been implemented in 30 seconds - we've spent more time writing comments than it would have taken to add a filter to the return, or even just add "Discussion" to the return string...

**Pull request merged**